shared/actions/chat/attachment.js

```diff
@@ -45,10 +53,20 @@ function * _isCached (conversationIDKey, messageID): Generator<any, ?string, any
   }
 }
 
-// TODO load previews too
 function * onLoadAttachment ({payload: {conversationIDKey, messageID, loadPreview, isHdPreview, filename}}: Constants.LoadAttachment): SagaGenerator<any, any> {
+  const existingMessage = yield select(Shared.messageSelector, conversationIDKey, messageID)
+  const existingMessageState = existingMessage && existingMessage.messageState
+
   // See if we already have this image cached
   if (loadPreview || isHdPreview) {
+    if (existingMessageState === 'downloading-preview') {
+      return
```

Comment:
Can we remove this? it can result in never having hd preview downloaded. Without it you might get progress jankiness, but at least it will eventually work